src/Mod/Part/Gui/DlgScale.cpp

```diff
@@ -0,0 +1,363 @@
+ *   Copyright (c) 2023 Wanderer Fan <wandererfan@gmail.com>               *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+# include <BRepAdaptor_Curve.hxx>
+# include <BRep_Tool.hxx>
+# include <Precision.hxx>
+# include <ShapeExtend_Explorer.hxx>
+# include <TopExp_Explorer.hxx>
+# include <TopoDS.hxx>
+# include <TopTools_HSequenceOfShape.hxx>
+# include <QKeyEvent>
+# include <QMessageBox>
+#include <QTreeWidget>
+#include <QTreeWidgetItem>
+#endif
+
+#include <App/Application.h>
+#include <App/Document.h>
+#include <App/DocumentObject.h>
+#include <App/Link.h>
+#include <App/Part.h>
+#include <Base/UnitsApi.h>
+#include <Gui/Application.h>
+#include <Gui/BitmapFactory.h>
+#include <Gui/Command.h>
+#include <Gui/Document.h>
+#include <Gui/Utilities.h>
+#include <Gui/ViewProvider.h>
+#include <Gui/WaitCursor.h>
+
+#include "ui_DlgScale.h"
+#include "DlgScale.h"
+
+
+FC_LOG_LEVEL_INIT("Part",true,true)
+
+using namespace PartGui;
+
+DlgScale::DlgScale(QWidget* parent, Qt::WindowFlags fl)
+  : QDialog(parent, fl), ui(new Ui_DlgScale)
+{
+    ui->setupUi(this);
+    setupConnections();
+
+    ui->dsbUniformScale->setDecimals(Base::UnitsApi::getDecimals());
+    ui->dsbXScale->setDecimals(Base::UnitsApi::getDecimals());
+    ui->dsbYScale->setDecimals(Base::UnitsApi::getDecimals());
+    ui->dsbZScale->setDecimals(Base::UnitsApi::getDecimals());
+    findShapes();
+}
+
+/*
+ *  Destroys the object and frees any allocated resources
+ */
+DlgScale::~DlgScale()

```

Comment:
According to the rules to modernize C++ code this should be written as: `DlgScale::~DlgScale() = default;`
Alternatively, you can completely drop the destructor.

Reply:
fixed